pkg/recipes/driver/terraform.go

```diff
@@ -135,13 +135,28 @@ func (d *terraformDriver) Delete(ctx context.Context, opts DeleteOptions) error
 			logger.Info(fmt.Sprintf("Failed to cleanup Terraform execution directory %q. Err: %s", requestDirPath, err.Error()))
 		}
 	}()
-
+	// Add credential information to .gitconfig if module source is of type git.
+	if strings.HasPrefix(opts.Definition.TemplatePath, "git::") && !reflect.DeepEqual(opts.BaseOptions.Secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
+		err := addSecretsToGitConfig(opts.BaseOptions.Secrets, &opts.Recipe, opts.Definition.TemplatePath)
+		if err != nil {
+			return err
+		}
+	}
 	err = d.terraformExecutor.Delete(ctx, terraform.Options{
 		RootDir:        requestDirPath,
 		EnvConfig:      &opts.Configuration,
 		ResourceRecipe: &opts.Recipe,
 		EnvRecipe:      &opts.Definition,
 	})
+
+	// Unset credential information from .gitconfig if module source is of type git.
+	if strings.HasPrefix(opts.Definition.TemplatePath, "git::") && !reflect.DeepEqual(opts.BaseOptions.Secrets, v20231001preview.SecretStoresClientListSecretsResponse{}) {
```

Comment:
Was this something caught during functional testing? Could you please update the PR description to include things that are being changed here in addition to adding functional test?

Reply:
I see this is being used a few times. We can make this a function itself.

Reply:
yes! this was caught during functional testing during clean up phase. Will update the PR description

Reply:
@ytimocin this refactoring will done as part of supporting `rad show` command

Reply:
> yes! this was caught during functional testing during clean up phase. Will update the PR description

I see. What issue was it causing? Complete failure to delete or issues with execution of next recipe request?

Reply:
complete failure of delete! as secrets were not added to delete path